modules/dnn/src/caffe/caffe_io.cpp

```diff
@@ -1130,7 +1130,17 @@ bool ReadProtoFromTextFile(const char* filename, Message* proto) {
     parser.AllowUnknownField(true);
     parser.SetRecursionLimit(1000);
 #endif
-    return parser.Parse(&input, proto);
+    const bool ret = parser.Parse(&input, proto);
+
+#ifdef OPENCV_DNN_EXTERNAL_PROTOBUF
+    if(!ret)
+    {
+        LOG(ERROR) << "Some data requires patched protobuf (available in OpenCV source tree only).";
+        CV_Error_(Error::StsError,("Some data requires patched protobuf (available in OpenCV source tree only)."));
```

Comment:
Return value might be false even with patched protobuf. So I recommend to keep it as is.

Reply:
Thank you for the review! I would like to revert it.

Currently, there is no way to distinguish between (1) and (2).

- (1) data that can be processed by only internal protobuf.
- (2) data that cannot be processed by both internal and external protobuf.

-----
I think this patch we can know which protobuf the issue reporter is used.

If an external unpatched protobuf, we can suggest trying the internal protobuf.
If a patched internal protobuf, we can focus our investigation on data content, etc.

However this is only workaround, so I agree with you.

Reply:
I reverted it, thank you for your comment !